giskardpy/src/giskardpy/motion_statechart/tasks/ros_tasks.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass, field
+
+import rclpy
+from geometry_msgs.msg import PoseStamped, Pose, Point, Quaternion
+from nav2_msgs.action import NavigateToPose
+from rclpy.action import ActionClient
+from std_msgs.msg import Header
+from typing_extensions import Any
+
+from giskardpy.motion_statechart.context import ExecutionContext, BuildContext
+from giskardpy.motion_statechart.graph_node import (
+    MotionStatechartNode,
+    NodeArtifacts,
+)
+from semantic_digital_twin.spatial_types import TransformationMatrix
+from semantic_digital_twin.world_description.world_entity import Body
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ActionServerTask(MotionStatechartNode):
+    """
+    Motion node for a ROS action server. Allows to send a goal to an action server while being part of a motion statechart.
+    """
+
+    action_topic: str
+    """
+    Topic name for the action server.
+    """
+
+    goal_msg: Any
+    """
+    Fully specified goal message that can be send out. 
+    """
+
+    node_handle: rclpy.node.Node
+    """
+    A ROS node to create the action client.
+    """
+
+    _action_client: ActionClient = field(init=False)
+    """
+    ROS action client, is created in `on_start`.
+    """
+
+    def build(self, context: BuildContext) -> NodeArtifacts:
+        """
+        Creates the action client.
+        """
+        self._action_client = ActionClient(
+            self.node_handle, self.goal_msg.__class__, self.action_topic
+        )
+        return NodeArtifacts()
+
+    def on_start(self, context: ExecutionContext):
+        logger.info(f"Waiting for action server {self.action_topic}")
+        self._action_client.wait_for_server()
+        logger.debug("Sending goal to action server")
+        future = self._action_client.send_goal_async(self.goal_msg)
+        future.add_done_callback(self.result_callback)
```

Comment:
implement `on_tick`. it should check if we have received a result, without blocking of course. if there is no result return unknown, on success True and on failure False